dialogs/primitives.py

```diff
@@ -1,55 +1,68 @@
 from typing import List, Any
+from dataclasses import dataclass
 
 from .types import Dialog, ClientResponse, RunSubdialog, message, send_to_client
 
 
-def prompt(text) -> Dialog:
-    def _prompt(run: RunSubdialog) -> ClientResponse:
-        run(message(text))
+@dataclass(frozen=True)
+class prompt(Dialog):
+    text: str
+    version = "1.0"
```

Comment:
Versions should be tuples to be easily comparable.

Reply:
I'm afraid this will mess up serialization, I'm sticking to trivial data types for now

Reply:
so list

Reply:
After some thought I'm keeping them as strings, maybe I'm old fashioned but "1.0" is easier to read than [1, 0]. I'm not sure I'll ever need an ordering, but if so I"ll deal with it.